pallets/order-book/src/lib.rs

```diff
@@ -582,6 +592,98 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Fill an existing order, based on the provided ratio.
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::Weights::fill_order_partial())]
+		pub fn fill_order_partial(
+			origin: OriginFor<T>,
+			order_id: T::OrderIdNonce,
+			fulfillment_ratio: Perquintill,
```

Comment:
~~On a second thought (I know I mentioned using `Perquintill`, so my bad for changing sides now) I think it would be easier to simply use `Balance` here. No rounding, easier for UI for get the "right" value.~~

Forget what I said. It is easier to handle both `sell_amount` and `buy_amount` with that, right? Then lets keep it.

Reply:
EDIT: I would actually still use `Balance` and manually convert the `buy_amount` into a `sell_amount` like we are already doing in `fn inner_update_order`. My main reasoning is that we have no rounding and that the UI can show exactly how much a user will need to "pay".

I.e. using `fn convert_with_ratio`. Its more expensive as we need two additional storage calls, but I hope its worth having not to deal with rounding. 

But of course make a case against that if you think it should be different.

Reply:
On second thought, I also agree that's the better option. This way we can ditch the `min_fulfillment_amount` update and also be super specific about the exact balance that we are buying, the UI could in turn have a feature to use `Perquintill` go get the actual balance, if needed.

Gonna update this, thank you for checking!

Reply:
Done.